app/main.py

```diff
@@ -19,16 +19,16 @@ def __add__(self, other: Distance | int | float) -> Distance:
     def __iadd__(self, other: Distance | int | float) -> Distance:
         if not isinstance(other, Distance):
             self.km += other
-            return self
-        self.km += other.km
+        else:
+            self.km += other.km
         return self
 
     def __mul__(self, other: int | float) -> Distance:
-        if isinstance(other, int) or isinstance(other, float):
+        if isinstance(other, (int, float)):
             return Distance(self.km * other)
 
     def __truediv__(self, other: int | float) -> Distance:
-        if (isinstance(other, int) or isinstance(other, float)) and other != 0:
+        if isinstance(other, (int, float)) and other != 0:
             return Distance(round(self.km / other, 2))
```

Comment:
You didn't fix it. `other` can not be of type `Distance` in this method, so you don't need to check for int/float types